internal/dataplane/translator/translate_upstreams.go

```diff
@@ -78,7 +78,7 @@ func (t *Translator) getUpstreams(serviceMap map[string]kongstate.Service) ([]ko
 				serviceMap[serviceName] = service
 
 				// get the new targets for this backend service
-				newTargets := getServiceEndpoints(t.logger, t.storer, k8sService, port, t.clusterDomain)
+				newTargets := getServiceEndpoints(t.logger, t.storer, k8sService, port, t.clusterDomain, t.featureFlags.StickySessionsTerminatingEndpoints)
```

Comment:
the feature gate should drive the feature enablement, not how the feature works. i would expect a new cli-argument (for example: `--drain-support`) and this parameter will be the real enabler. the feature gate can disappear in future (because promoted in GA and set `true` by default) and in that case i will miss an option to drive the enablement

Reply:
After communication, we reached an agreement to use `--enable-drain-support` for control and remove the feature gate.